tests/dragonfly/replication_test.py

```diff
@@ -1836,3 +1836,83 @@ async def test_replicaof_reject_on_load(df_local_factory, df_seeder_factory):
     await c_replica.close()
     master.stop()
     replica.stop()
+
+
+# note: please be careful if you want to change any of the parameters used in this test.
+# changing parameters without extensive testing may easily lead to weak testing case assertion
+# which means eviction may not get triggered.
+@pytest.mark.asyncio
+@pytest.mark.slow
+async def test_policy_based_eviction_propagation(df_local_factory, df_seeder_factory):
+    master = df_local_factory.create(
+        proactor_threads=1, cache_mode="true", maxmemory="256mb", enable_heartbeat_eviction="false"
+    )
+    replica = df_local_factory.create(proactor_threads=1)
+    df_local_factory.start_all([master, replica])
+
+    c_master = master.client()
+    c_replica = replica.client()
+
+    await c_replica.execute_command(f"REPLICAOF localhost {master.port}")
+    await wait_available_async(c_replica)
+
+    seeder = df_seeder_factory.create(
+        port=master.port, keys=100000, val_size=1000, stop_on_failure=False
+    )
+
+    # Filling data to trigger policy based eviction.
+    fill_task = asyncio.create_task(seeder.run())
+    while True:
+        time.sleep(0.1)
+        info = await c_master.info("stats")
+        if info["evicted_keys"] > 0:
+            break
+    seeder.stop()
+    fill_task.cancel()
+
+    # Check data after stable state stream
+    await check_all_replicas_finished([c_replica], c_master)
+    await check_data(seeder, [replica], [c_replica])
+    await disconnect_clients(c_master, *[c_replica])
+
+
+@pytest.mark.asyncio
+async def test_heartbeat_eviction_propagation(df_local_factory):
```

Comment:
@adiholden i just split the previous test into two separate tests. This newly added one only test the HB based eviction. (The previous one now only tests policy based eviction) 

The HB eviction test can be tested using debug populate, and can be done within seconds.